services/posts.service.go

```diff
@@ -0,0 +1,69 @@
+package services
+
+import (
+	"context"
+	"time"
+
+	repositories "postsql/repositories/postgres"
+)
+
+type CreatePostParams struct {
+	Title     string    `json:"title"`
+	Content   string    `json:"content"`
+	User      string    `json:"user"`
+	CreatedAt time.Time `json:"created_at,omitempty"`
+	UpdatedAt time.Time `json:"updated_at,omitempty"`
+	Image     string    `json:"image,omitempty"`
+}
+
+type Post struct {
+	UID       string    `json:"uid"`
+	Title     string    `json:"title"`
+	Content   string    `json:"content"`
+	User      string    `json:"user"`
+	CreatedAt time.Time `json:"created_at"`
+	UpdatedAt time.Time `json:"updated_at"`
+	Image     string    `json:"image,omitempty"`
```

Comment:
so you have json tags but not sql ones ? (note I don't know sqlx etc but I would expect it has its own tags?)

Reply:
> so you have json tags but not sql ones ? (note I don't know sqlx etc but I would expect it has its own tags?)

I have a separate struct called `PostModel` for the DB, just didn't want to import model struct into the service layer. Not sure if this makes any sense, though.